### PR TITLE
Dockerfile: update syntax, switch to bullseye, add missing libseccomp-dev, remove build pack

### PR DESCRIPTION
**Pull request by thaJeztah**
So, this  started with the intention to "just" update `buster` to `bullseye`, but finding various issues that needed fixing, or could be improved.

### Dockerfile: update to docker/dockerfile:1.3, and remove temporary fix.

I saw we were using an older syntax, and the issue I reported (https://github.com/moby/buildkit/issues/2114) was fixed in dockerfile:1.3 front-end, so upgrading allowed me to remove the temporary fix.

### Dockerfile: remove aufs-tools, as it's not available on bullseye

Well, title says all. No more aufs?

### Dockerfile: update to debian bullseye

Well, that's what  I came here for 😂 

### Dockerfile: add back libseccomp-dev to cross-compile runc

Commit https://github.com/moby/moby/commit/7168d98c434af0a35e8c9a05dfb87bb40511a38a removed these, but I _think_ we overlooked that the same stage is used to build runc as well, so we likely need these. (but happy to remove if we really don't need them!)

### Dockerfile: frozen images: update to bullseye, remove buildpack-dep

Update the frozen images to also be based on Debian bullseye. Using the "slim" variant (which looks to have all we're currently using),  and remove the buildpack-dep frozen image.

The buildpack-dep image is quite large, and it looks like we only use it to compile some C binaries, which should work fine on a regular debian image;

    docker build -t debian:bullseye-slim-gcc -<<EOF
    FROM debian:bullseye-slim
    RUN apt-get update && apt-get install -y gcc libc6-dev --no-install-recommends
    EOF

    docker image ls

    REPOSITORY        TAG                 IMAGE ID       CREATED          SIZE
    debian            bullseye-slim-gcc   1851750242af   About a minute ago   255MB
    buildpack-deps    bullseye            fe8fece98de2   2 days ago           834MB

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: integration/build/build_userns_linux_test.go [hunk 1]
```diff
@@ -53,7 +53,8 @@ func TestBuildUserNamespaceValidateCapabilitiesAreV2(t *testing.T) {
 	}()
 
 	dockerfile := `
-		FROM debian:bullseye
+		FROM debian:bullseye-slim
+		RUN apt-get update && apt-get install -y libcap2-bin --no-install-recommends
```
Comment (thaJeztah) [inline integration/build/build_userns_linux_test.go:57]: Looks like `setcap` was removed from bullseye at some point;

```bash
docker run --rm debian:bullseye@sha256:7190e972ab16aefea4d758ebe42a293f4e5c5be63595f4d03a5b9bf6839a4344  sh -c 'setcap CAP_NET_BIND_SERVICE=+eip /bin/sleep'

docker run --rm debian:bullseye  sh -c 'setcap CAP_NET_BIND_SERVICE=+eip /bin/sleep'
sh: 1: setcap: not found

**Comment by thaJeztah** [comment]
Arf.. output format changed of `getcap`; 

```
archive_unix_test.go:288: assertion failed: string "/tmp/docker-test-untar-origin629714680/2 cap_block_suspend=ep\n" does not contain "= cap_block_suspend+ep": untar should have kept the 'security.capability' xattr
```

**Comment by thaJeztah** [comment]
Ahm... needs some more tests to be updated;

```
=== RUN   TestDaemonDefaultNetworkPools
    service_test.go:23: assertion failed:
        Command:  ip link delete docker0
        ExitCode: 127
        Error:    exec: "ip": executable file not found in $PATH
        Stdout:
        Stderr:

        Failures:
        ExitCode was 127 expected 0
        Expected no error

=== RUN   TestDaemonRestartWithExistingNetwork
    service_test.go:23: assertion failed: 
        Command:  ip link delete docker0
        ExitCode: 127
        Error:    exec: "ip": executable file not found in $PATH
        Stdout:   
        Stderr:   
        
        Failures:
        ExitCode was 127 expected 0
        Expected no error
```

**Comment by thaJeztah** [comment]
Getting close now; only one related failure;

```
=== RUN   TestDockerSuite/TestAPIStatsNetworkStats
    docker_api_stats_test.go:145: assertion failed: error is not nil: exec: "ping": executable file not found in $PATH
    --- FAIL: TestDockerSuite/TestAPIStatsNetworkStats (1.75s)
```

File: Dockerfile [hunk 2]
```diff
@@ -254,9 +253,10 @@ RUN --mount=type=cache,sharing=locked,id=moby-dev-aptlib,target=/var/lib/apt \
     --mount=type=cache,sharing=locked,id=moby-dev-aptcache,target=/var/cache/apt \
         apt-get update && apt-get install -y --no-install-recommends \
             apparmor \
-            aufs-tools \
             bash-completion \
             bzip2 \
+            inetutils-ping\
```
Comment (tianon) [inline Dockerfile:258]:
```suggestion
            inetutils-ping \
```

:eyes:

File: Dockerfile.e2e [hunk 3]
```diff
@@ -66,7 +66,9 @@ RUN apk --no-cache add \
     ca-certificates \
     g++ \
     git \
+    inetutils-ping\
```
Comment (tianon) [inline Dockerfile.e2e:69]:
```suggestion
    inetutils-ping \
```

File: Dockerfile (hunk 2 above)
Comment (thaJeztah) [reply Dockerfile:258]: 🙈

**Comment by thaJeztah** [comment]
@tianon fixed; PTAL

**Comment by thaJeztah** [comment]
New flaky test? ("interesting" error)

```
=== RUN   TestServicePlugin
    plugin_test.go:50: assertion failed: error is not nil: Error response from daemon: content sha256:fc6183ae6e37a7808e1ee317bb5d513b52d0053577e973606e9725f26135ac7d: already exists
--- FAIL: TestServicePlugin (2.09s)

File: Dockerfile [hunk 4]
```diff
@@ -21,11 +21,9 @@ ENV GO111MODULE=off
 FROM base AS criu
 ARG DEBIAN_FRONTEND
 ADD --chmod=0644 https://download.opensuse.org/repositories/devel:/tools:/criu/Debian_10/Release.key /etc/apt/trusted.gpg.d/criu.gpg.asc
-# FIXME: temporarily doing a manual chmod as workaround for https://github.com/moby/buildkit/issues/2114
 RUN --mount=type=cache,sharing=locked,id=moby-criu-aptlib,target=/var/lib/apt \
     --mount=type=cache,sharing=locked,id=moby-criu-aptcache,target=/var/cache/apt \
-        chmod 0644 /etc/apt/trusted.gpg.d/criu.gpg.asc \
-        && echo 'deb https://download.opensuse.org/repositories/devel:/tools:/criu/Debian_10/ /' > /etc/apt/sources.list.d/criu.list \
+        echo 'deb https://download.opensuse.org/repositories/devel:/tools:/criu/Debian_10/ /' > /etc/apt/sources.list.d/criu.list \
```
Comment (tianon) [inline Dockerfile:26]: Should we add something like `TODO update to Debian_11` or just not worry about it?  (It's probably fine.)

**Review by tianon — APPROVED** [review]

File: Dockerfile (hunk 4 above)
Comment (thaJeztah) [reply Dockerfile:26]: I'm hoping @kolyshkin will remind me at that time; https://github.com/opencontainers/runc/pull/3171#discussion_r692213776 😁

**Comment by thaJeztah** [comment]
@AkihiroSuda @cpuguy83 PTAL

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**